test/component/uk/gov/hmrc/individualsincomeapi/TaxYearIntervalValidationSpec.scala

```diff
@@ -0,0 +1,115 @@
+
+      When("I request individual SA income with a missing fromTaxYear")
+      val response = Http(s"$serviceUrl/sandbox/sa?matchId=$sandboxMatchId&toTaxYear=2016-17")
+        .headers(requestHeaders(acceptHeaderP1)).asString
+
+      Then("The response status should be 400 (Bad Request)")
+      response.code shouldBe BAD_REQUEST
+
+      And("The correct error message is returned")
+      response.body shouldBe errorResponse("fromTaxYear is required")
+    }
+
+    scenario("invalid format for fromTaxYear parameter submitted") {
+
+      When("I request individual SA income with an incorrectly formatted fromTaxYear")
+      val response = Http(s"$serviceUrl/sandbox/sa?matchId=$sandboxMatchId&fromTaxYear=20162017&toTaxYear=2016-17")
+        .headers(requestHeaders(acceptHeaderP1)).asString
+
+      Then("The response status should be 400 (Bad Request)")
+      response.code shouldBe BAD_REQUEST
+
+      And("The correct error message is returned")
+      response.body shouldBe errorResponse("fromTaxYear: invalid tax year format")
+    }
+
+    scenario("invalid format for toTaxYear parameter submitted") {
+
+      When("I request individual SA income with an incorrectly formatted toTaxYear")
+      val response = Http(s"$serviceUrl/sandbox/sa?matchId=$sandboxMatchId&fromTaxYear=2016-17&toTaxYear=20170301")
+        .headers(requestHeaders(acceptHeaderP1)).asString
+
+      Then("The response status should be 400 (Bad Request)")
+      response.code shouldBe BAD_REQUEST
+
+      And("The correct error message is returned")
+      response.body shouldBe errorResponse("toTaxYear: invalid tax year format")
+    }
+
+    scenario("invalid interval submitted. toTaxYear before fromTaxYear") {
+
+      When("I request individual income with toTaxYear value before fromTaxYear")
+      val response = Http(s"$serviceUrl/sandbox/sa?matchId=$sandboxMatchId&fromTaxYear=2016-17&toTaxYear=2015-16")
+        .headers(requestHeaders(acceptHeaderP1)).asString
+
+      Then("The response status should be 400 (Bad Request)")
+      response.code shouldBe BAD_REQUEST
+
+      And("The correct error message is returned")
+      response.body shouldBe errorResponse("Invalid time period requested")
+    }
+
+    scenario("fromTaxYear earlier than maximum allowed") {
+
+      When("I request individual income with fromTaxYear 7 years before the current tax year")
+      val fromTaxYear = TaxYear.fromEndYear(today.getYear - 8)
+      val response = Http(s"$serviceUrl/sandbox/sa?matchId=$sandboxMatchId&fromTaxYear=${fromTaxYear.formattedTaxYear}&toTaxYear=2015-16")
+        .headers(requestHeaders(acceptHeaderP1)).asString
+
+      Then("The response status should be 400 (Bad Request)")
+      response.code shouldBe BAD_REQUEST
+
+      And("The correct error message is returned")
+      response.body shouldBe errorResponse("fromTaxYear earlier than maximum allowed")
+    }
+
+    scenario("toTaxYear defaults to today's date when it is not provided") {
```

Comment:
shouldn't this be 'defaults to the next available tax year after fromTaxYear'?
And if so, then what happens if fromTaxYear is 2016-17 and no toTaxYear is provided?

Reply:
done.
toTaxYear currently default to 2017-18.